src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotReferSelfInSerializableClass.cs

```diff
@@ -66,14 +66,14 @@ public sealed override void Initialize(AnalysisContext context)
                         return;
                     }
 
-                    var forwardGraph = new ConcurrentDictionary<ISymbol, ConcurrentDictionary<ISymbol, bool>>();
-                    var invertedGraph = new ConcurrentDictionary<ISymbol, ConcurrentDictionary<ISymbol, bool>>();
+                    ConcurrentDictionary<ISymbol, ConcurrentDictionary<ISymbol, bool>> forwardGraph = new ConcurrentDictionary<ISymbol, ConcurrentDictionary<ISymbol, bool>>();
```

Comment:
Please retarget all security rule changes to 2.9.x branch. Also, there is no nullability involved here, so why was this required?

Reply:
It's not practical to backport this change. The 2.9.x branch doesn't support this toolchain or compiler version so it would be a large backport of much more than just this pull request.

I don't believe these changes will be a problem in practice. /cc @dotpaul for awareness.